public/src/managers/CourseManager.ts

```diff
@@ -309,6 +312,29 @@ export class CourseManager {
         return await this.courseProvider.getCourseGroups(courseid);
     }
 
+    /**
+     * Load an IGroupCourse object for a single group and a single course
+     * @param group The group the information should be retrived from
+     * @param course The course the data should be loaded for
+     */
+    public async getGroupCourse(group: ICourseGroup, course: ICourse): Promise<IGroupCourse | null> {
```

Comment:
Naming of the first input argument: `ICourseGroup` is misleading (just `group`)?

Also, it is a bit confusing to have both `ICourseGroup` and `IGroupCourse`. I assume this is useful and there is a reason for it.

Reply:
Yes, this is the same problem as the one mentioned above, I think a refactor is necessary to avoid confusing and to be more clear.